File: FragileBaseClassProblem/BankAccount.cs
class BankAccount
{
    protected double balance;
    public virtual void Deposit(double amount)
    {
        balance += amount;
        LogTransaction(amount);
    }
    protected virtual void LogTransaction(double amount)
    {
        Console.WriteLine($"Deposited: {amount}");
    }
}
class AuditAccount : BankAccount
{
    protected override void LogTransaction(double amount)
    {
        Console.WriteLine($"Audit log: {amount}");
    }
}

File: FragileBaseClassProblem/BankAccount.ts
class BankAccount {
    protected balance: number = 0;

    deposit(amount: number): void {
        this.balance += amount;
        this.logTransaction(amount);
    }

    protected logTransaction(amount: number): void {
        console.log(`Deposited: ${amount}`);
    }
}

class AuditAccount extends BankAccount {
    protected logTransaction(amount: number): void {
        console.log(`Audit log: ${amount}`);
    }
}
